migrations/20200717111739-mapping_referance_project_to_private_program.js

```diff
@@ -0,0 +1,111 @@
+var fs = require("fs");
+let _ = require("lodash");
+
+module.exports = {
+  async up(db) {
+
+    global.migrationMsg = "Mapping referance project to private program"
+
+    let usersProjects = await db.collection('userProjects').aggregate([
+      {
+        $match: { "createdType": { $eq: "by reference" } }
+      },
+      {
+        $group: {
+          _id: { userId: { userId: "$userId" } },
+          project: { $push: { _id: "$_id", solutionId: "$solutionId", programId: "$programId" } }
+        }
+      }
+    ]).toArray();
+
+    
+    let usersProjectsArray = _.chunk(usersProjects, 2);
+    async function checkPrivateProgramExist(userId) {
+
+      let programDoc = await db.collection('programs').findOne({ owner: userId }, { _id: 1 });
```

Comment:
@rakeshSgr  Can we not check a filter for isAPrivateProgram ??

Reply:
the code is for old database structure,  Do we need this check ?

Reply:
@rakeshSgr  Yes